packages/docid/src/commit-id.ts

```diff
@@ -0,0 +1,225 @@
+ */
+export class CommitID implements DocRef {
+  readonly #doctype: number;
+  readonly #cid: CID;
+  readonly #commit: CID | null; // null ≝ genesis commit
+
+  static fromBytes = fromBytes;
+  static fromString = fromString;
+
+  /**
+   * Create a new DocID.
+   *
+   * @param {string|number}      doctype
+   * @param {CID|string}         cid
+   * @param {CID|string}         commit CID. Pass '0', 0, or omit the value as shorthand for the genesis commit.
+   *
+   * @example
+   * new DocID(<docType>, <CID>|<cidStr>)
+   * new DocID(<docType>, <CID>|<cidStr>, <CommitCID>|<CommitCidStr>)
+   */
+  constructor(doctype: string | number, cid: CID | string, commit: CID | string | number = null) {
+    if (!doctype && doctype !== 0) throw new Error('constructor: doctype required');
+    if (!cid) throw new Error('constructor: cid required');
+    this.#doctype = typeof doctype === 'string' ? doctypes.indexByName(doctype) : doctype;
+    this.#cid = typeof cid === 'string' ? new CID(cid) : cid;
+    this.#commit = parseCommit(this.#cid, commit);
+  }
+
+  /**
+   * Construct DocID, no commit information included
+   */
+  get baseID(): DocID {
+    return new DocID(this.#doctype, this.#cid);
+  }
+
+  /**
+   * Doc type code
+   */
+  get type(): number {
+    return this.#doctype;
+  }
+
+  /**
+   * Doc type name
+   */
+  get typeName(): string {
+    return doctypes.nameByIndex(this.#doctype);
+  }
+
+  /**
+   * Genesis CID
+   */
+  get cid(): CID {
+    return this.#cid;
+  }
+
+  /**
+   * Commit CID
+   */
+  get commit(): CID {
+    return this.#commit || this.#cid;
+  }
+
+  /**
+   * Bytes representation
+   */
+  @Memoize()
+  get bytes(): Uint8Array {
+    const codec = varint.encode(DOCID_CODEC);
+    const doctype = varint.encode(this.type);
+
+    const commitBytes = this.#commit?.bytes || new Uint8Array([0]);
+    return uint8ArrayConcat([codec, doctype, this.cid.bytes, commitBytes]);
+  }
+
+  /**
+   * Construct new CommitID for the same document, but a new `commit` CID.
+   */
+  travel(commit: CID | string | number): CommitID {
+    return new CommitID(this.#doctype, this.#cid, commit);
+  }
+
+  /**
+   * Compare equality with another CommitID.
+   */
+  equals(other: CommitID): boolean {
+    return this.type === other.type && this.cid.equals(other.cid) && this.commit.equals(other.commit);
+  }
+
+  /**
+   * Encode the CommitID into a string.
+   */
+  toString(): string {
+    return uint8ArrayToString(multibase.encode(DEFAULT_BASE, this.bytes));
+  }
+
+  /**
+   * Encode the DocID into a base36 url
+   */
+  toUrl(): string {
+    return `ceramic://${this.toString()}`;
+  }
+
+  /**
+   * If genesis commit:
+   * CommitID(k3y52l7mkcvtg023bt9txegccxe1bah8os3naw5asin3baf3l3t54atn0cuy98yws, 0)
+   *
+   * If commit:
+   * CommitID(k3y52l7mkcvtg023bt9txegccxe1bah8os3naw5asin3baf3l3t54atn0cuy98yws, bagjqcgzaday6dzalvmy5ady2m5a5legq5zrbsnlxfc2bfxej532ds7htpova)
```

Comment:
Can we please output this as an encoded CommitID? 
I.e. this.toString().

Reply:
The point of this inspection is to provide an engineer with a hint about document identifier and commit. IMO two-part output is more comprehensible while triage. And if not triage, I do not see why the inspection is ever needed. Do you think that one-part makes it easier to read?

Reply:
Summoning @stbrody and @valmack as consumers of the API here I have a suspicion, two-part display here makes sense, but as I do not consume the API (yet), I have little to say.

Reply:
> if a debugger was attached

yes.